scripts/gen-lvms-kustomize.sh

```diff
@@ -0,0 +1,182 @@
+# otherwise, only the operator is deployed through OLM
+LVMS_CLUSTER_CR=${LVMS_CLUSTER_CR:-0}
+
+# Timeout defaults
+API_TIMEOUT=100
+CSV_TIMEOUT=300
+TIME=5
+
+# Disk map in the form <disk_name>:<size>
+declare -A DEVICES
+DEVICES["vdb"]=100
+
+if [ ! -d "${DEPLOY_DIR}" ]; then
+    mkdir -p "${DEPLOY_DIR}"
+fi
+
+pushd "${DEPLOY_DIR}"
+
+# Add a device passed as input and attach it to crc
+function add_device {
+    local disk_name="$1"
+    local size="${2-$DISK_SIZE}"
+    if [[ ! -e "${TARGET_DIR}/$disk_name" ]]; then
+        echo "Creating Device $disk_name:$size"
+        sudo -S qemu-img create -f raw "${TARGET_DIR}/${disk_name}" "${size}"G
+    fi
+    # this condition might fail if the disk is attached but not present in the
+    # virsh dumpxml output
+    if [[ -z "$(sudo virsh dumpxml crc | grep "$disk_name")" ]]; then
+        # Attach the disk to the crc VM: it requires reboot
+        echo "Attaching disk $disk_name to crc"
+        sudo virsh attach-disk crc "${TARGET_DIR}/${disk_name}" "${disk_name}" --config
+        # Trigger a CRC restart
+        REQUIRE_RESTART=1
+    fi
+}
+
+# Restart CRC if REQUIRE_RESTART=1: this is required
+# to make the new devices available
+function restart_crc {
+    sudo virsh destroy crc
+    sleep "$TIME"
+    sudo virsh start crc
+    status=""
+    while [ -z $status ]; do
+        if curl -k https://"${OCP_API}":6443; then
+            status="ok";
+        else
+            sleep "$TIME"
+        fi
+    done
+}
+
+# Deploy the LVMS operator based on the official OLM deploy samples
+function lvms_operator_kustomize {
+cat <<EOF >kustomization.yaml
+resources:
+- https://github.com/openshift/lvm-operator/config/olm-deploy
+EOF
+
+# apply the LVMS operator
+oc kustomize "${DEPLOY_DIR}" | oc apply -f -
+}
+
+# By default we're going to use all the available
+# disks provided by crc
+function lvms_cluster_deploy {
+cat <<EOF >lvms_cluster.yaml
+apiVersion: lvm.topolvm.io/v1alpha1
+kind: LVMCluster
+metadata:
+  name: lvmcluster
+spec:
+  storage:
+    deviceClasses:
+      - name: vg1
+        fstype: ext4
+        default: true
+        thinPoolConfig:
+          name: thin-pool-1
+          sizePercent: 90
+          overprovisionRatio: 10
+EOF
+
+# apply the resulting CR
+oc apply -f "${DEPLOY_DIR}"/lvms_cluster.yaml
+}
+
+
+# Patch openshift-storage namespace to get the right annotations
+function patch_lvms_ns {
+cat <<EOF >lvms_ns.yaml
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: openshift-storage
+  annotations:
+    workload.openshift.io/allowed: "management"
+    openshift.io/sa.scc.uid-range: "1000090000/10000"
+    openshift.io/sa.scc.mcs: "s0:c26,c5"
+  labels:
+    pod-security.kubernetes.io/enforce: "privileged"
+    openshift.io/cluster-monitoring: "false"
+EOF
+
+# apply the LVMS ns
+oc apply -f "${DEPLOY_DIR}"/lvms_ns.yaml
+}
+
+function patch_csv_metrics {
+    until oc get csv lvms-operator.v0.0.1 &> /dev/null; do
```

Comment:
Do you want to add a `-n` to the `oc get csv` command?

```
[fultonj@hamfast install_yamls{lvms}]$ oc get csv lvms-operator.v0.0.1
Error from server (NotFound): clusterserviceversions.operators.coreos.com "lvms-operator.v0.0.1" not found
[fultonj@hamfast install_yamls{lvms}]$ oc -n openshift-storage get csv lvms-operator.v0.0.1
NAME                   DISPLAY       VERSION   REPLACES   PHASE
lvms-operator.v0.0.1   LVM Storage   0.0.1                Succeeded
[fultonj@hamfast install_yamls{lvms}]$ 
```

Reply:
In my case this was still hanging until I added it. 
```
[fultonj@hamfast install_yamls{lvms}]$ make lvms
bash scripts/gen-lvms-kustomize.sh
~/install_yamls/out/openstack/lvms/cr ~/install_yamls
Creating Device vdb:100
Formatting '/home/fultonj/.crc/vdb', fmt=raw size=107374182400
Attaching disk vdb to crc
Disk attached successfully

Domain 'crc' destroyed

Domain 'crc' started

curl: (7) Failed to connect to 192.168.130.11 port 6443: No route to host
curl: (7) Failed to connect to 192.168.130.11 port 6443: Connection refused
The apiserver hasn't been fully initialized yet, please try again later.
Wait for the OCP API to be fully available
......
namespace/openshift-storage created
operatorgroup.operators.coreos.com/openshift-storage-operatorgroup created
catalogsource.operators.coreos.com/lvms-catalogsource created
subscription.operators.coreos.com/lvms-operator created
Rolling out the lvms-operator
namespace/openshift-storage configured
................................................................^Cmake: *** [Makefile:2031: lvms] Interrupt

[fultonj@hamfast install_yamls{lvms}]$ vim scripts/gen-lvms-kustomize.sh
[fultonj@hamfast install_yamls{lvms}]$ make lvms
bash scripts/gen-lvms-kustomize.sh
~/install_yamls/out/openstack/lvms/cr ~/install_yamls
Wait for the OCP API to be fully available

namespace/openshift-storage configured
operatorgroup.operators.coreos.com/openshift-storage-operatorgroup unchanged
catalogsource.operators.coreos.com/lvms-catalogsource unchanged
subscription.operators.coreos.com/lvms-operator unchanged
Rolling out the lvms-operator
namespace/openshift-storage configured

clusterserviceversion.operators.coreos.com/lvms-operator.v0.0.1 patched
[fultonj@hamfast install_yamls{lvms}]$ 
```

Reply:
makes sense. The `CSV` is a namespaced resource so we should always point to what is happening within `openshift-storage`. I'll patch this part in my next iteration.

Reply:
Done, also, I added the namespace in the `lvmcluster` object to make sure we don't create a resource that is not seen by the lvms-operator.